src/mds/Server.cc

```diff
@@ -3450,6 +3450,10 @@ void Server::handle_client_getattr(MDRequestRef& mdr, bool is_lookup)
 
   // reply
   dout(10) << "reply to stat on " << *req << dendl;
+  if (mask & CEPH_STAT_DIR_PIN) {
+    dout(20) << " encode inode export_pin=" << ref->inode.export_pin << dendl;
+    encode(ref->inode.export_pin, mdr->reply_extra_bl);
```

Comment:
This should go in `CInode::encode_inodestat` I think. Then just increment the encoded stat version.

Reply:
I thought of this at the beginning, but this required to add a CEPH FEATURE BIT and check client session do has this feature before encoding export_pin value. So I switched to use a flag for VXattr.

Reply:
@batrick  Oh, I see what you mean now. The latest master branch is so different from Luminous version at `CInode::encode_inodestat`. I will make the changes accordingly.